src/bin/cargo/commands/doc.rs

```diff
@@ -51,11 +62,36 @@ the `cargo help pkgid` command.
 
 pub fn exec(config: &mut Config, args: &ArgMatches<'_>) -> CliResult {
     let ws = args.workspace(config)?;
-    let mode = CompileMode::Doc {
-        deps: !args.is_present("no-deps"),
-    };
-    let mut compile_opts =
-        args.compile_options(config, mode, Some(&ws), ProfileChecking::Checked)?;
+
+    let mut deps_mode = DocDepsMode::default();
+    if let Some(deps) = args.values_of("deps") {
+        if !config.cli_unstable().unstable_options {
+            Err(anyhow::anyhow!(
+                "Usage of `--deps` requires `-Z unstable-options`"
+            ))?;
```

Comment:
There is a function for doing this:

```rust
config.cli_unstable().fail_if_stable_opt("--deps", 1234)?;
```

Can you open a tracking issue and use the issue number?